core/src/main/java/org/elasticsearch/index/analysis/SynonymTokenFilterFactory.java

```
@@ -23,82 +23,151 @@
     private final boolean ignoreCase;
 
-    public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, Map<String, TokenizerFactory> tokenizerFactories,
-                                      String name, Settings settings) throws IOException {
+    public SynonymTokenFilterFactory(IndexSettings indexSettings, Environment env, AnalysisRegistry analysisRegistry,
+                                     Map<String, TokenizerFactory> tokenizerFactories, String name, Settings settings) throws IOException {
         super(indexSettings, name, settings);
 
-        Reader rulesReader = null;
         if (settings.getAsArray("synonyms", null) != null) {
-            List<String> rules = Analysis.getWordList(env, settings, "synonyms");
+            List<String> rulesList = Analysis.getWordList(env, settings, "synonyms");
             StringBuilder sb = new StringBuilder();
-            for (String line : rules) {
+            for (String line : rulesList) {
                 sb.append(line).append(System.lineSeparator());
             }
-            rulesReader = new FastStringReader(sb.toString());
+            rules = sb.toString();
         } else if (settings.get("synonyms_path") != null) {
-            rulesReader = Analysis.getReaderFromFile(env, settings, "synonyms_path");
+            Reader rulesReader = Analysis.getReaderFromFile(env, settings, "synonyms_path");
+            BufferedReader br = new BufferedReader(rulesReader);
+            StringBuilder sb = new StringBuilder();
+            String tmp;
+            try {
+                while ((tmp = br.readLine()) != null) {
+                    sb.append(tmp).append(System.getProperty("line.separator"));
+                }
+                rules = sb.toString();
+            } catch (IOException ioe) {
+                throw new IllegalArgumentException("failed to load synonyms", ioe);
+            } finally {
+                IOUtils.closeWhileHandlingException(br);
```

I don't think we should swallow the exceptions here, instead, this could be a try-with-resources block like:

```java
try (BufferedReader br = new BufferedReader(rulesReader)) {
    .. read the string ..
}
```